### PR TITLE
Implement Beacon node fork rest api

### LINKED ISSUE #409: Implement optional get fork rest endpoint
https://app.swaggerhub.com/apis/spble/beacon_node_api_for_validator/0.2.0#/OptionalSet/get_node_fork

### PR DESCRIPTION
**Pull request by twoeths**
## Goal
+ Close #409

### CODE REVIEW

File: packages/lodestar/src/api/rpc/api/beacon/beacon.ts [hunk 1]
```diff
@@ -31,9 +31,13 @@ export class BeaconApi implements IBeaconApi {
     return Buffer.from(`lodestar-${process.env.npm_package_version}`, 'utf-8');
   }
 
-  public async getFork(): Promise<Fork> {
+  public async getFork(): Promise<{fork: Fork; chain_id: uint64}> {
     const state: BeaconState = await this.db.state.getLatest();
-    return state.fork;
+    const networkId: uint64 = await this.chain.networkId;
```
Comment (mpetrunic) [inline packages/lodestar/src/api/rpc/api/beacon/beacon.ts]: could you create `{fork: BeaconState, chainId: number64}` in impl directory and use it from both rest and rpc so we don't have duplicate code

File: packages/lodestar/src/api/rpc/api/beacon/interface.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@ export interface IBeaconApi extends IApi {
   /**
    * Requests the BeaconNode to provide which fork version it is currently on.
    */
-  getFork(): Promise<Fork>;
+  getFork(): Promise<{fork: Fork; chain_id: uint64}>;
```
Comment (mpetrunic) [inline packages/lodestar/src/api/rpc/api/beacon/interface.ts]: I would keep everything camelcase in rpc

**Review by mpetrunic — CHANGES_REQUESTED** [review]

**Review by mpetrunic — APPROVED** [review]

File: packages/lodestar/src/api/impl/beacon/fork.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { Fork, uint64, BeaconState } from "@chainsafe/eth2.0-types";
+import { IBeaconDb } from "../../../db";
+import { IBeaconChain } from "../../../chain";
+
+
+export async function getFork(db: IBeaconDb, chain: IBeaconChain): Promise<{fork: Fork; chainId: uint64}> {
+  const state: BeaconState = await db.state.getLatest();
+  const networkId: uint64 = await chain.networkId;
```
Comment (mpetrunic) [inline packages/lodestar/src/api/impl/beacon/fork.ts]: do we really need await here?

File: packages/lodestar/src/api/impl/beacon/fork.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { Fork, uint64, BeaconState } from "@chainsafe/eth2.0-types";
+import { IBeaconDb } from "../../../db";
+import { IBeaconChain } from "../../../chain";
+
+
+export async function getFork(db: IBeaconDb, chain: IBeaconChain): Promise<{fork: Fork; chainId: uint64}> {
+  const state: BeaconState = await db.state.getLatest();
```
Comment (mpetrunic) [inline packages/lodestar/src/api/impl/beacon/fork.ts:7]: niot sure if api specifies error in case chain didn't yet started (getLatest returns null)?

**Review by mpetrunic — CHANGES_REQUESTED** [review]
Also could you add test case when beacon state is null (chain didn't start)

**Review by mpetrunic — APPROVED** [review]

**Review by wemeetagain — APPROVED** [review]
lgtm
